internal/controller/operator_reconciliator.go

```diff
@@ -0,0 +1,114 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *OLSConfigReconciler) generateServiceMonitorForOperator() (*monv1.ServiceMonitor, error) {
+	metaLabels := map[string]string{
+		"control-plane":                              "controller-manager",
+		"app.kubernetes.io/component":                "metrics",
+		"app.kubernetes.io/managed-by":               "lightspeed-operator",
+		"app.kubernetes.io/name":                     "servicemonitor",
+		"app.kubernetes.io/instance":                 "controller-manager-metrics-monitor",
+		"app.kubernetes.io/part-of":                  "lightspeed-operator",
+		"monitoring.openshift.io/collection-profile": "full",
```

Comment:
@raptorsun we did not have this label in the past - do we need this ?

Reply:
This label is not neccessary.
This label will turn off metrics collections for this service when using a minimal collection profile on resource limited clusters.